src/components/pagination/OcPagination.vue

```diff
@@ -0,0 +1,337 @@
+<template>
+  <nav class="oc-pagination" :aria-label="$gettext('Pagination')">
+    <ul class="oc-pagination-list">
+      <li v-if="isPrevPageAvailable" class="oc-pagination-list-item">
+        <router-link
+          class="oc-pagination-list-item-prev"
+          :aria-label="$gettext('Go to the previous page')"
+          :to="previousPageLink"
+        >
+          <oc-icon name="chevron_left" />
+        </router-link>
+      </li>
+      <li v-for="(page, index) in displayedPages" :key="index" class="oc-pagination-list-item">
+        <component
+          :is="pageComponent(page)"
+          :class="pageClass(page)"
+          v-bind="pageProps(page)"
+          v-text="page"
+        />
+      </li>
+      <li v-if="isNextPageAvailable" class="oc-pagination-list-item">
+        <router-link
+          class="oc-pagination-list-item-next"
+          :aria-label="$gettext('Go to the next page')"
+          :to="nextPageLink"
+        >
+          <oc-icon name="chevron_right" />
+        </router-link>
+      </li>
+      <li v-if="isGoToLinkVisible" class="oc-pagination-list-item-goto">
+        <oc-text-input
+          v-model="goToTarget"
+          :label="$gettext('Go to')"
+          class="oc-pagination-list-item-goto-input"
+        />
+        <component
+          :is="goToLinkComponent"
+          v-bind="goToLinkProps"
+          class="oc-pagination-list-item-goto-link"
+        >
+          <oc-icon name="chevron_right" />
+        </component>
+      </li>
+    </ul>
+  </nav>
+</template>
+
+<script>
+import OcIcon from "../OcIcon.vue"
+
+/**
+ * A list of links used for switching to different pages
+ */
+export default {
+  name: "OcPagination",
+
+  components: { OcIcon },
+
+  props: {
+    /**
+     * Number of pages to be displayed
+     */
+    pages: {
+      type: Number,
+      required: true,
+    },
+    /**
+     * Currently active page
+     */
+    currentPage: {
+      type: Number,
+      required: true,
+    },
+    /**
+     * Number of pages to be displayed. Needs to be an even number.
```

Comment:
What's the reason for requiring an even number?

Reply:
That number is evenly split into both sides. If there's 4 max visible, 2 will be visible on each side of the current page. I wanted to have it originally as the number for those sides separately i.e. 2 = 2 on each side but I have gone with this way mainly due to not being able to come up with the correct naming.

Reply:
Ah, makes sense. But then it should enforced to be an uneven number - you always have the current page and want to display the same amount of pages on both sides. Which makes `x+x+1 = uneven`.

Reply:
In my head, I took the current page outside of this number. Which doesn't make any sense with the current wording. What about changing the docs to `Number of pages to be displayed next to the current page evenly split into both sides. It is required to use an even number.`

Reply:
IMO creates too much complexity. All paginations that I ever used treated this as the total number of displayed pages, including the current page. 😉

Reply:
... which is simple and doesn't require you to calculate `x - 1` in your head.